system/doc/design_principles/statem.xml

```diff
@@ -1912,10 +1912,16 @@ do_unlock() ->
       to implicitly postpone any events to the <c>locked</c> state.
     </p>
     <p>
-      A selective receive cannot be used from a <c>gen_statem</c>
+      A catch-all receive should never be used from a <c>gen_statem</c>
       behaviour (or from any <c>gen_*</c> behaviour),
       as the receive statement is within the <c>gen_*</c> engine itself.
-      It must be there because all
+      Using such a receive may result in system messages
+      being discarded which in turn may lead to unexpected behaviour.
+      If a selective receive must be used then great care should be taken to
+      ensure messages only messages pertinent to the operation are caught.
```

Comment:
First "messages" is redundant

Reply:
Btw, this phrasing still leaves a backdoor open, ie messages of the form `{system, ..., ...}` or `{'$gen_call', ..., ...}` _could_ be pertinent to the operation but would also be prone to stealing non-pertinent messages. However, I think that this is unlikely, and that you can't point out every single way in which someone could do something wrong. Just nitpicking then :sweat_smile:

Reply:
Yeah, but I agree. We're jamming a whole lot into a few sentences here That's why I figure a long-term plan might be to add a lot more to a section where longer explanations with lots of examples make sense. 

Either way, I've adjusted a few things, see if it read better now.